lib/urlwatch/reporters.py

```diff
@@ -123,35 +133,36 @@ def reporter_documentation(cls):
 
     @classmethod
     def submit_one(cls, name, report, job_states, duration):
+        any_enabled = False
         subclass = cls.__subclasses__[name]
-        cfg = report.config['report'].get(name, {'enabled': False})
-        if cfg['enabled']:
-            base_config = subclass.get_base_config(report)
-            if base_config.get('separate', False):
-                for job_state in job_states:
-                    subclass(report, cfg, [job_state], duration).submit()
-            else:
-                subclass(report, cfg, job_states, duration).submit()
-        else:
-            raise ValueError('Reporter not enabled: {name}'.format(name=name))
+        cfgs = report.config['report'].get(name, {'enabled': False})
+        if isinstance(cfgs, Mapping):
+            cfgs = [cfgs]
 
-    @classmethod
-    def submit_all(cls, report, job_states, duration):
-        any_enabled = False
-        for name, subclass in cls.__subclasses__.items():
-            cfg = report.config['report'].get(name, {})
+        for cfg in cfgs:
             if cfg.get('enabled', False):
                 any_enabled = True
                 logger.info('Submitting with %s (%r)', name, subclass)
                 base_config = subclass.get_base_config(report)
+                matching_job_states = filter_by_tags(job_states, cfg.get("tags", []))
                 if base_config.get('separate', False):
-                    for job_state in job_states:
-                        subclass(report, cfg, [job_state], duration).submit()
+                    for job_state in matching_job_states:
+                        subclass(report, cfg, [job_state], len(job_states), duration).submit()
+                        job_state.reported_count = job_state.reported_count + 1
                 else:
-                    subclass(report, cfg, job_states, duration).submit()
+                    subclass(report, cfg, matching_job_states, len(job_states), duration).submit()
+                    for job_state in matching_job_states:
+                        job_state.reported_count = job_state.reported_count + 1
+
+        return any_enabled
+
+    @classmethod
+    def submit_all(cls, report, job_states, duration):
+        any_enabled = False
+        for name in cls.__subclasses__.keys():
```

Comment:
This might also work:

```suggestion
        for name in cls.__subclasses__:
```